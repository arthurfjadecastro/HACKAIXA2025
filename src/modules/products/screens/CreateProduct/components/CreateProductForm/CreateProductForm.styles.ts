import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
  formContainer: {
    paddingHorizontal: 20, // Padding lateral para respiro visual
  },
  section: {
    marginBottom: 32, // Espaçamento aumentado entre seções
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: '600',
    color: '#333333',
    marginBottom: 12, // Espaçamento aumentado
  },
  sectionDescription: {
    fontSize: 14,
    color: '#667085',
    marginBottom: 20, // Espaçamento aumentado
    lineHeight: 20, // Melhor legibilidade
  },
  radioGroup: {
    gap: 16, // Gap aumentado entre opções
  },
  radioOption: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#FFFFFF',
    padding: 20, // Padding aumentado para melhor toque
    borderRadius: 12,
    borderWidth: 1,
    borderColor: '#E5E7EB',
    elevation: 1, // Sombra sutil no Android
    shadowColor: '#000', // Sombra no iOS
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.05,
    shadowRadius: 2,
  },
  radioOptionSelected: {
    borderColor: '#005CA9',
    backgroundColor: '#F0F7FF',
    elevation: 2, // Sombra maior quando selecionado
    shadowOpacity: 0.1,
  },
  radioOptionDisabled: {
    backgroundColor: '#F9FAFB',
    opacity: 0.6,
  },
  radioButton: {
    width: 20,
    height: 20,
    borderRadius: 10,
    borderWidth: 2,
    borderColor: '#D1D5DB',
    marginRight: 12,
    alignItems: 'center',
    justifyContent: 'center',
  },
  radioButtonSelected: {
    width: 10,
    height: 10,
    borderRadius: 5,
    backgroundColor: '#005CA9',
  },
  radioContent: {
    flex: 1,
  },
  radioLabel: {
    fontSize: 16,
    fontWeight: '600',
    color: '#333333',
    marginBottom: 4,
  },
  radioLabelDisabled: {
    color: '#9CA3AF',
  },
  radioDescription: {
    fontSize: 14,
    color: '#667085',
  },
  radioSubInfo: {
    fontSize: 12,
    color: '#059669',
    fontWeight: '500',
    marginTop: 4,
  },
  radioDescriptionDisabled: {
    color: '#D1D5DB',
  },
  autoFilledBadge: {
    fontSize: 12,
    color: '#059669',
    fontWeight: '400',
  },
  autoFilledContainer: {
    backgroundColor: '#F0FDF4',
    padding: 16,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: '#D1FAE5',
  },

  // Container destacado para informações principais
  highlightedInfoContainer: {
    backgroundColor: '#ffffff',
    borderWidth: 2,
    borderColor: '#28a745',
    borderRadius: 8,
    padding: 12,
    marginBottom: 16,
    shadowColor: '#000',
    shadowOffset: { width: 0, height: 2 },
    shadowOpacity: 0.1,
    shadowRadius: 3,
    elevation: 3,
  },

  highlightedInfoRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    paddingVertical: 8,
    borderBottomWidth: 1,
    borderBottomColor: '#e8f5e8',
  },

  highlightedInfoLabel: {
    fontSize: 14,
    fontWeight: '600',
    color: '#2c3e50',
  },

  highlightedInfoValue: {
    fontSize: 16,
    fontWeight: '700',
    color: '#28a745',
  },

  // Container para as taxas
  taxasContainer: {
    backgroundColor: '#ffffff',
    borderWidth: 1,
    borderColor: '#dee2e6',
    borderRadius: 6,
    padding: 10,
  },

  taxasTitle: {
    fontSize: 13,
    fontWeight: '600',
    color: '#495057',
    marginBottom: 8,
    textAlign: 'center',
  },

  autoFilledRow: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginBottom: 8,
  },
  autoFilledLabel: {
    fontSize: 14,
    color: '#374151',
    fontWeight: '500',
  },
  autoFilledValue: {
    fontSize: 14,
    color: '#059669',
    fontWeight: '600',
  },
  observacoesContainer: {
    marginTop: 16,
    backgroundColor: '#FEF3C7',
    padding: 16,
    borderRadius: 12,
    borderWidth: 1,
    borderColor: '#FCD34D',
  },
  observacoesTitle: {
    fontSize: 14,
    fontWeight: '600',
    color: '#92400E',
    marginBottom: 8,
  },
  observacaoItem: {
    fontSize: 13,
    color: '#92400E',
    marginBottom: 4,
  },

  // Estilos para normativo legal
  normativoContainer: {
    backgroundColor: '#EEF2FF',
    borderWidth: 1,
    borderColor: '#C7D2FE',
    borderRadius: 8,
    padding: 12,
  },

  normativoText: {
    fontSize: 14,
    fontWeight: '600',
    color: '#3730A3',
    marginBottom: 4,
  },

  normativoDescription: {
    fontSize: 12,
    color: '#6366F1',
    fontStyle: 'italic',
  },

  // Estilos específicos para formulário OUTRO
  inputGroup: {
    marginBottom: 16,
  },
  label: {
    fontSize: 14,
    fontWeight: '600',
    color: '#374151',
    marginBottom: 8,
  },
  inputContainer: {
    backgroundColor: '#F9FAFB',
    borderWidth: 1,
    borderColor: '#E5E7EB',
    borderRadius: 8,
    padding: 12,
  },
  input: {
    fontSize: 14,
    color: '#111827',
  },
  checkboxGroup: {
    gap: 12,
  },
  checkboxItem: {
    flexDirection: 'row',
    alignItems: 'center',
    gap: 12,
  },
  checkbox: {
    width: 20,
    height: 20,
    borderWidth: 2,
    borderColor: '#D1D5DB',
    borderRadius: 4,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: '#FFFFFF',
  },
  checkboxChecked: {
    backgroundColor: '#005CA9',
    borderColor: '#005CA9',
  },
  checkmark: {
    color: '#FFFFFF',
    fontSize: 12,
    fontWeight: 'bold',
  },
  checkboxLabel: {
    fontSize: 14,
    color: '#374151',
    flex: 1,
  },
  rowContainer: {
    flexDirection: 'row',
    gap: 16,
    marginBottom: 16,
  },
  halfWidth: {
    flex: 1,
  },
  fieldHint: {
    fontSize: 12,
    color: '#6B7280',
    marginTop: 4,
    fontStyle: 'italic',
  },
  warningContainer: {
    backgroundColor: '#FEF2F2',
    borderWidth: 1,
    borderColor: '#FECACA',
    borderRadius: 8,
    padding: 12,
    marginBottom: 16,
  },
  warningText: {
    fontSize: 14,
    color: '#DC2626',
    fontWeight: '500',
    textAlign: 'center',
  },
});
