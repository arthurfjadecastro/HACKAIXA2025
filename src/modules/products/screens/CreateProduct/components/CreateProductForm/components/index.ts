export { CategorySelector } from './CategorySelector';
export { SubtypeSelector } from './SubtypeSelector';
export { ConvenioSelector } from './ConvenioSelector';
export { AutoFilledSummary } from './AutoFilledSummary';
export { OutroProductFields } from './OutroProductFields';
