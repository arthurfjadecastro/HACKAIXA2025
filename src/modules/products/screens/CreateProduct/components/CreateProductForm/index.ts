export { CreateProductForm } from './CreateProductForm.refactored';
export * from './components';
export * from './types';
