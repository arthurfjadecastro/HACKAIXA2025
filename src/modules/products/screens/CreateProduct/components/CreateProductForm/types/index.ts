import { FormData, FieldName } from '../../../types';

export interface CreateProductFormProps {
  formData: FormData;
  updateField: (fieldName: FieldName, value: string | number | boolean | string[]) => void;
  isConvenioAlreadyRegistered: (convenioKey: string) => boolean;
  isHabitacaoAlreadyRegistered: () => boolean;
}

export interface CategoryOption {
  value: string;
  label: string;
  description: string;
  disabled?: boolean;
}
