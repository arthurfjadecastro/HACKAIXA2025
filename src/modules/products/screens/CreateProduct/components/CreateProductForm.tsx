import React from 'react';
import { View } from 'react-native';
import { Text } from '@/design-system/components';

// Importar componentes modulares
import {
  CategorySelector,
  SubtypeSelector,
  ConvenioSelector,
  AutoFilledSummary,
  OutroProductFields,
} from './CreateProductForm/components';

import { styles } from './CreateProductForm/CreateProductForm.styles';
import { CreateProductFormProps } from './CreateProductForm/types';

export const CreateProductForm: React.FC<CreateProductFormProps> = ({
  formData,
  updateField,
  isConvenioAlreadyRegistered,
  isHabitacaoAlreadyRegistered,
}) => {
  return (
    <View style={styles.formContainer}>
      {/* Seletor de Categoria */}
      <CategorySelector
        selectedCategory={formData.categoria}
        onCategorySelect={(categoria: string) => updateField('categoria', categoria)}
        isHabitacaoAlreadyRegistered={isHabitacaoAlreadyRegistered}
      />

      {/* Seletor de Subtipo */}
      <SubtypeSelector
        categoria={formData.categoria}
        selectedSubtipo={formData.subtipo}
        onSubtipoSelect={(subtipo: string) => updateField('subtipo', subtipo)}
        isConvenioAlreadyRegistered={isConvenioAlreadyRegistered}
      />

      {/* Seletor de Convênio */}
      <ConvenioSelector
        subtipo={formData.subtipo}
        selectedConvenio={formData.convenio_selected || ''}
        onConvenioSelect={(convenio: string) => updateField('convenio_selected', convenio)}
        isConvenioAlreadyRegistered={isConvenioAlreadyRegistered}
      />

      {/* Resumo dos dados preenchidos automaticamente */}
      <AutoFilledSummary formData={formData} />

      {/* Normativo Legal - Se foi preenchido automaticamente */}
      {(formData.categoria === 'CONSIGNADO' || formData.categoria === 'HABITACAO') && formData.normative && (
        <View style={styles.section}>
          <Text style={styles.sectionTitle}>Normativo Legal</Text>
          
          <View style={styles.normativoContainer}>
            <Text style={styles.normativoText}>{formData.normative}</Text>
            <Text style={styles.normativoDescription}>
              {formData.categoria === 'CONSIGNADO' 
                ? 'Normativo aplicável para produtos de crédito consignado'
                : 'Normativo aplicável para financiamento habitacional'
              }
            </Text>
          </View>
        </View>
      )}

      {/* Campos manuais para categoria OUTRO */}
      <OutroProductFields 
        formData={formData} 
        updateField={updateField} 
      />

      {/* Por enquanto, produtos específicos (CONSIGNADO/HABITACAO) são configurados automaticamente */}
    </View>
  );
};
