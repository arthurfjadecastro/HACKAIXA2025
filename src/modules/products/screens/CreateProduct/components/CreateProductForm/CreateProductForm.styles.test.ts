import { styles } from './CreateProductForm.styles';

describe('CreateProductForm Styles', () => {
  it('should export styles object', () => {
    expect(styles).toBeDefined();
    expect(typeof styles).toBe('object');
  });

  it('should have formContainer style', () => {
    expect(styles.formContainer).toBeDefined();
    expect(styles.formContainer.paddingHorizontal).toBe(20);
  });

  it('should have section style', () => {
    expect(styles.section).toBeDefined();
    expect(styles.section.marginBottom).toBe(32);
  });

  it('should have sectionTitle style', () => {
    expect(styles.sectionTitle).toBeDefined();
    expect(styles.sectionTitle.fontSize).toBe(18);
    expect(styles.sectionTitle.fontWeight).toBe('600');
    expect(styles.sectionTitle.color).toBe('#333333');
    expect(styles.sectionTitle.marginBottom).toBe(12);
  });

  it('should have sectionDescription style', () => {
    expect(styles.sectionDescription).toBeDefined();
    expect(styles.sectionDescription.fontSize).toBe(14);
    expect(styles.sectionDescription.color).toBe('#667085');
    expect(styles.sectionDescription.marginBottom).toBe(20);
    expect(styles.sectionDescription.lineHeight).toBe(20);
  });

  it('should have radioGroup style', () => {
    expect(styles.radioGroup).toBeDefined();
    expect(styles.radioGroup.gap).toBe(16);
  });

  it('should have radioOption style', () => {
    expect(styles.radioOption).toBeDefined();
    expect(styles.radioOption.flexDirection).toBe('row');
    expect(styles.radioOption.alignItems).toBe('center');
    expect(styles.radioOption.backgroundColor).toBe('#FFFFFF');
    expect(styles.radioOption.padding).toBe(20);
    expect(styles.radioOption.borderRadius).toBe(12);
  });

  it('should match snapshot', () => {
    expect(styles).toMatchSnapshot();
  });
});
