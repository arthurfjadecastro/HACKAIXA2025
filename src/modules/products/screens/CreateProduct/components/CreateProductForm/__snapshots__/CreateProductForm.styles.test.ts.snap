// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateProductForm Styles should match snapshot 1`] = `
{
  "autoFilledBadge": {
    "color": "#059669",
    "fontSize": 12,
    "fontWeight": "400",
  },
  "autoFilledContainer": {
    "backgroundColor": "#F0FDF4",
    "borderColor": "#D1FAE5",
    "borderRadius": 12,
    "borderWidth": 1,
    "padding": 16,
  },
  "autoFilledLabel": {
    "color": "#374151",
    "fontSize": 14,
    "fontWeight": "500",
  },
  "autoFilledRow": {
    "alignItems": "center",
    "flexDirection": "row",
    "justifyContent": "space-between",
    "marginBottom": 8,
  },
  "autoFilledValue": {
    "color": "#059669",
    "fontSize": 14,
    "fontWeight": "600",
  },
  "checkbox": {
    "alignItems": "center",
    "backgroundColor": "#FFFFFF",
    "borderColor": "#D1D5DB",
    "borderRadius": 4,
    "borderWidth": 2,
    "height": 20,
    "justifyContent": "center",
    "width": 20,
  },
  "checkboxChecked": {
    "backgroundColor": "#005CA9",
    "borderColor": "#005CA9",
  },
  "checkboxGroup": {
    "gap": 12,
  },
  "checkboxItem": {
    "alignItems": "center",
    "flexDirection": "row",
    "gap": 12,
  },
  "checkboxLabel": {
    "color": "#374151",
    "flex": 1,
    "fontSize": 14,
  },
  "checkmark": {
    "color": "#FFFFFF",
    "fontSize": 12,
    "fontWeight": "bold",
  },
  "fieldHint": {
    "color": "#6B7280",
    "fontSize": 12,
    "fontStyle": "italic",
    "marginTop": 4,
  },
  "formContainer": {
    "paddingHorizontal": 20,
  },
  "halfWidth": {
    "flex": 1,
  },
  "highlightedInfoContainer": {
    "backgroundColor": "#ffffff",
    "borderColor": "#28a745",
    "borderRadius": 8,
    "borderWidth": 2,
    "elevation": 3,
    "marginBottom": 16,
    "padding": 12,
    "shadowColor": "#000",
    "shadowOffset": {
      "height": 2,
      "width": 0,
    },
    "shadowOpacity": 0.1,
    "shadowRadius": 3,
  },
  "highlightedInfoLabel": {
    "color": "#2c3e50",
    "fontSize": 14,
    "fontWeight": "600",
  },
  "highlightedInfoRow": {
    "alignItems": "center",
    "borderBottomColor": "#e8f5e8",
    "borderBottomWidth": 1,
    "flexDirection": "row",
    "justifyContent": "space-between",
    "paddingVertical": 8,
  },
  "highlightedInfoValue": {
    "color": "#28a745",
    "fontSize": 16,
    "fontWeight": "700",
  },
  "input": {
    "color": "#111827",
    "fontSize": 14,
  },
  "inputContainer": {
    "backgroundColor": "#F9FAFB",
    "borderColor": "#E5E7EB",
    "borderRadius": 8,
    "borderWidth": 1,
    "padding": 12,
  },
  "inputGroup": {
    "marginBottom": 16,
  },
  "label": {
    "color": "#374151",
    "fontSize": 14,
    "fontWeight": "600",
    "marginBottom": 8,
  },
  "normativoContainer": {
    "backgroundColor": "#EEF2FF",
    "borderColor": "#C7D2FE",
    "borderRadius": 8,
    "borderWidth": 1,
    "padding": 12,
  },
  "normativoDescription": {
    "color": "#6366F1",
    "fontSize": 12,
    "fontStyle": "italic",
  },
  "normativoText": {
    "color": "#3730A3",
    "fontSize": 14,
    "fontWeight": "600",
    "marginBottom": 4,
  },
  "observacaoItem": {
    "color": "#92400E",
    "fontSize": 13,
    "marginBottom": 4,
  },
  "observacoesContainer": {
    "backgroundColor": "#FEF3C7",
    "borderColor": "#FCD34D",
    "borderRadius": 12,
    "borderWidth": 1,
    "marginTop": 16,
    "padding": 16,
  },
  "observacoesTitle": {
    "color": "#92400E",
    "fontSize": 14,
    "fontWeight": "600",
    "marginBottom": 8,
  },
  "radioButton": {
    "alignItems": "center",
    "borderColor": "#D1D5DB",
    "borderRadius": 10,
    "borderWidth": 2,
    "height": 20,
    "justifyContent": "center",
    "marginRight": 12,
    "width": 20,
  },
  "radioButtonSelected": {
    "backgroundColor": "#005CA9",
    "borderRadius": 5,
    "height": 10,
    "width": 10,
  },
  "radioContent": {
    "flex": 1,
  },
  "radioDescription": {
    "color": "#667085",
    "fontSize": 14,
  },
  "radioDescriptionDisabled": {
    "color": "#D1D5DB",
  },
  "radioGroup": {
    "gap": 16,
  },
  "radioLabel": {
    "color": "#333333",
    "fontSize": 16,
    "fontWeight": "600",
    "marginBottom": 4,
  },
  "radioLabelDisabled": {
    "color": "#9CA3AF",
  },
  "radioOption": {
    "alignItems": "center",
    "backgroundColor": "#FFFFFF",
    "borderColor": "#E5E7EB",
    "borderRadius": 12,
    "borderWidth": 1,
    "elevation": 1,
    "flexDirection": "row",
    "padding": 20,
    "shadowColor": "#000",
    "shadowOffset": {
      "height": 1,
      "width": 0,
    },
    "shadowOpacity": 0.05,
    "shadowRadius": 2,
  },
  "radioOptionDisabled": {
    "backgroundColor": "#F9FAFB",
    "opacity": 0.6,
  },
  "radioOptionSelected": {
    "backgroundColor": "#F0F7FF",
    "borderColor": "#005CA9",
    "elevation": 2,
    "shadowOpacity": 0.1,
  },
  "radioSubInfo": {
    "color": "#059669",
    "fontSize": 12,
    "fontWeight": "500",
    "marginTop": 4,
  },
  "rowContainer": {
    "flexDirection": "row",
    "gap": 16,
    "marginBottom": 16,
  },
  "section": {
    "marginBottom": 32,
  },
  "sectionDescription": {
    "color": "#667085",
    "fontSize": 14,
    "lineHeight": 20,
    "marginBottom": 20,
  },
  "sectionTitle": {
    "color": "#333333",
    "fontSize": 18,
    "fontWeight": "600",
    "marginBottom": 12,
  },
  "taxasContainer": {
    "backgroundColor": "#ffffff",
    "borderColor": "#dee2e6",
    "borderRadius": 6,
    "borderWidth": 1,
    "padding": 10,
  },
  "taxasTitle": {
    "color": "#495057",
    "fontSize": 13,
    "fontWeight": "600",
    "marginBottom": 8,
    "textAlign": "center",
  },
  "warningContainer": {
    "backgroundColor": "#FEF2F2",
    "borderColor": "#FECACA",
    "borderRadius": 8,
    "borderWidth": 1,
    "marginBottom": 16,
    "padding": 12,
  },
  "warningText": {
    "color": "#DC2626",
    "fontSize": 14,
    "fontWeight": "500",
    "textAlign": "center",
  },
}
`;
